File: backend/src/main/java/com/x/pojo/entity/ChatMessage.java
package com.x.pojo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ChatMessage implements Serializable {
    private Long id;
    private Long sessionId;
    private Long userId;
    private String content;
    private Role role;  // user / ai

    public enum Role {
        user,
        ai
    }
}

File: frontend/src/api/chatMessage.ts
import request from "@/utils/request";

 export interface ChatMessage {
  id: number
  sessionId: number
  userId: number 
  role: 'user' | 'ai'
  content: string
  isWaiting: boolean 
}



function getCurMsgList(sessionId: number){
  return request({
    url: `/ai/getCurMsgList/${sessionId}`,
    method: 'GET',
  })
}

function addMsg(msg: ChatMessage){
  return request({
    url: `/ai/addMsg`,
    method: 'POST',
    data: msg
  })
}

function deleteMsg(msgId: number){
  return request({
    url: `/ai/deleteMsg/${msgId}`,
    method: 'DELETE',
  })
}

export default {
  getCurMsgList,
  addMsg,
  deleteMsg
}